### PR TITLE
[receiver/pulsar] Change the types of `Token` and `PrivateKey` to be `configopaque.String`

### PR DESCRIPTION
**Pull request by mackjmr**
**Description:**
Split out from: #17353

**Link to tracking Issue:** #17273

### CODE REVIEW

**Review by atoulme — APPROVED** [review]

**Review by codeboten — COMMENTED** [review]
Pinging code owners @dmitryax @tjiuming

File: .chloggen/pulsar-use-configopaque.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# Use this changelog template to create an entry for release notes.
+# If your change doesn't affect end users, such as a test fix or a tooling change,
+# you should instead start your pull request title with [chore] or use the "Skip Changelog" label.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: breaking
```
Comment (codeboten) [inline .chloggen/pulsar-use-configopaque.yaml:6]: to be clear this is a breaking changes to users of the code as a library, not to end users correct?

Comment (mackjmr) [reply .chloggen/pulsar-use-configopaque.yaml:6]: Yes, that is correct.

**Review by tjiuming — APPROVED** [review]
LGTM

**Review by dmitryax — APPROVED** [review]

**Pull request merged**